### DISCUSSION TITLE
Latenz

### DISCUSSION THREAD

**Discussion by FireFly3390** [camueller/SmartApplianceEnabler#435] [Installation und Betrieb]
Hallo zusammen,

ich beobachte seit Beginn der Nutzung von SAE, dass die Latenz allgemein sehr hoch und scheinbar auch nicht konstant ist.
Schalte ich testweise einen Verbraucher ein und kontrolliere parallel den aktuellen Verbrauch über das Sunny Portal, so fällt eine Latenz zwischen 30 Sekunden und 1,5 min auf. Mir ist bewusst, dass alleine die Verarbeitung über die SMA Server ihre Zeit dauert und auch nicht unbedingt konstant ist, jedoch kommen Werte, welche die Fritz DECT Dose misst deutlich schneller im Portal an (grob 10 - 20 s). Auch der durch den Home Manager direkt gemessene Verbrauch/Bezug kommt deutlich schneller an.
Das ist zwar kein Beinbruch aber als technikaffiner Mensch frage ich mich, ob das schneller geht, bzw. wodurch diese Latenz begründet ist. Speziell bei der PV geführten Steuerung von Verbrauchern schadet eine geringere Latenz nie.

Wer weiß dazu mehr?

Gruß

**Comment by camueller** [comment]
Der SHM fragt den SAE im Normalfall nur all 60s ab. Dadurch ist Latenz systembedingt gegeben. In der Praxis spielt das aber keine Rolle.

**Reply by FireFly3390** [reply]
Verstehe, ich nehme an man kann den SHM nicht dazu bewegen, die abfragen schneller/häufiger zu machen. In der Zählerkonfiguration des SHM konnte ich nur eine Einstellung zum Messintervall des internen Energy Meter finden.
Über welche Schnittstelle kommuniziert der SHM denn mit der Fritzbox? SEMP ja offensichtlich nicht... Eingabe der Zugangsdaten deutet ja auch auf eine Art Webhook oder WebAPI hin. Eventuell kann man sich da mit dran hängen.
Ist aber vermutliches alles zu viel Arbeit um ein nicht wirklich vorhandenes Problem zu lösen^^

**Comment by Doenerteller** [comment]
Ich fände ein kürzeres Abfrageinterval des SHM beim SAE auch nicht übel. Ich habe zwar noch kein echtes Problem mit der Minute festgestellt, aber es fühlt sich für mich einfach zu grob an. Gerade wenn ein Verbraucher vielleicht innerhalb einer Minute nicht konstant die gleiche Leistung abruft. Oder nur relativ kurz für weniger als eine Minute richtig "Gas gibt". Dann kommt der SHM vielleicht dann erst vorbei, wenn der Verbraucher schon wieder fertig ist. Dieser Verbrauch würde gar nicht erfasst. Zugegeben, das sind am Ende bestimmt nur minimale "verpasste" Verbräuche. Die Minute fühlt sich jedoch nicht wirklich toll an und über die Zeit summiert sich das bestimmt auf. Wie gesagt, ich habe kein Gefühl dafür ob es ein überhaupt ein Problem sein könnte und ob es solche Verbraucher wie von mir beschrieben gibt, die immer genau in die Abfragelücken hauen. 
Ist den meisten Leuten bestimmt auch egal.

Entnehme ich der Formulierung "...im Normalfall" korrekt, dass man es auch umstellen könnte und öfter abfragen und richtung SMA schicken könnte? Oder macht das Backend bei SMA aus feineren Meßdaten am Ende doch wieder nur grobe Hackschnitzel und es ist daher egal?

Im SAE kann man das Abfrageintervall beim Zähler einstellen. Auch auf unter 60s. Wozu eigenlich, wenn der SHM dann doch nur alle Minute daher kommt?

**Comment by camueller** [comment]
> Gerade wenn ein Verbraucher vielleicht innerhalb einer Minute nicht konstant die gleiche Leistung abruft. Oder nur relativ kurz für weniger als eine Minute richtig "Gas gibt". Dann kommt der SHM vielleicht dann erst vorbei, wenn der Verbraucher schon wieder fertig ist.

Deshalb beim Zähler  lieber Zählertände abfragen als Leistungswerte.

> Die Minute fühlt sich jedoch nicht wirklich toll an und über die Zeit summiert sich das bestimmt auf. Wie gesagt, ich habe kein Gefühl dafür ob es ein überhaupt ein Problem sein könnte und ob es solche Verbraucher wie von mir beschrieben gibt, die immer genau in die Abfragelücken hauen.

Wir haben kürzlich die [Genauigkeit der Tagessummen im Portal betrachtet](https://github.com/camueller/SmartApplianceEnabler/discussions/427) und zumindest bei Verbrauchern mit hoher Leistungsaufnahme und qualitativ guten Zählern (keine Steckdosenadapter), haben die Zahlen sehr gut übereingestimmt.

> Entnehme ich der Formulierung "...im Normalfall" korrekt, dass man es auch umstellen könnte und öfter abfragen und richtung SMA schicken könnte? Oder macht das Backend bei SMA aus feineren Meßdaten am Ende doch wieder nur grobe Hackschnitzel und es ist daher egal?

Das Sunny Portal ist ein kostenloses Angebot von SMA. Ich denke, dass ihr da einfach zu viel erwartet. Datensätze mit höherer Auflösung als 1/Minute sind teuer und haben in der Praxis keine Relevanz.

> Im SAE kann man das Abfrageintervall beim Zähler einstellen. Auch auf unter 60s. Wozu eigenlich, wenn der SHM dann doch nur alle Minute daher kommt?

Um einen besseren Wert bei Geräten mit schwankender Leistungsaufnahme zu haben bzw. schalten sich die Geräte ja nicht nur zur vollen Minute ein und aus :-)

**Comment by Doenerteller** [comment]
Danke für's feedback!
